framework/src/executioners/FixedPointSolve.C

```diff
@@ -0,0 +1,565 @@
+  if (_old_entering_time == _problem.time())
+    savePostprocessorValues(false);
+
+  if (converged)
+  {
+    // Update the subapp using the fixed point algorithm
+    if (_secondary_transformed_variables.size() > 0 &&
+        useFixedPointAlgorithmUpdateInsteadOfPicard(false) && _old_entering_time == _problem.time())
+      transformVariables(secondary_transformed_dofs, false);
+
+    // Update the entering time, used to detect failed solves
+    _old_entering_time = _problem.time();
+  }
+
+  if (_has_fixed_point_its)
+    printFixedPointConvergenceReason();
+
+  return converged;
+}
+
+void
+FixedPointSolve::saveAllValues(const bool primary)
+{
+  saveVariableValues(primary);
+  savePostprocessorValues(primary);
+}
+
+bool
+FixedPointSolve::solveStep(Real & begin_norm,
+                           Real & end_norm,
+                           const std::set<dof_id_type> & transformed_dofs)
+{
+  bool auto_advance = autoAdvance();
+
+  // Compute previous norms for coloring the norm output
+  Real begin_norm_old = (_fixed_point_it > 0 ? _fixed_point_timestep_begin_norm[_fixed_point_it - 1]
+                                             : std::numeric_limits<Real>::max());
+  Real end_norm_old = (_fixed_point_it > 0 ? _fixed_point_timestep_end_norm[_fixed_point_it - 1]
+                                           : std::numeric_limits<Real>::max());
+
+  _executioner.preSolve();
+
+  _problem.execTransfers(EXEC_TIMESTEP_BEGIN);
+  if (!_problem.execMultiApps(EXEC_TIMESTEP_BEGIN, auto_advance))
+  {
+    _fixed_point_status = MooseFixedPointConvergenceReason::DIVERGED_FAILED_MULTIAPP;
+    return false;
+  }
+
+  if (_problem.haveXFEM() && _update_xfem_at_timestep_begin)
+    _problem.updateMeshXFEM();
+
+  _problem.execute(EXEC_TIMESTEP_BEGIN);
+
+  // Transform the fixed point postprocessors before solving, but after the timestep_begin transfers
+  // have been received
+  if (_transformed_pps.size() > 0 && useFixedPointAlgorithmUpdateInsteadOfPicard(true))
+    transformPostprocessors(true);
+  if (_secondary_transformed_pps.size() > 0 && useFixedPointAlgorithmUpdateInsteadOfPicard(false) &&
+      _problem.time() == _old_entering_time)
+    transformPostprocessors(false);
+
+  if (_has_fixed_point_its && _has_fixed_point_norm)
+    if (_problem.hasMultiApps(EXEC_TIMESTEP_BEGIN) || _fixed_point_force_norms)
+    {
+      begin_norm = _problem.computeResidualL2Norm();
+
+      _console << COLOR_MAGENTA << "Fixed point residual norm after TIMESTEP_BEGIN MultiApps: "
+               << Console::outputNorm(begin_norm_old, begin_norm) << '\n';
+    }
+
+  // Perform output for timestep begin
+  _problem.outputStep(EXEC_TIMESTEP_BEGIN);
+
+  // Update warehouse active objects
+  _problem.updateActiveObjects();
+
+  // Save the current values of variables and postprocessors, before the solve
+  saveAllValues(true);
+
+  if (_has_fixed_point_its)
+    _console << COLOR_MAGENTA << "\nMain app solve:\n" << COLOR_DEFAULT;
```

Comment:
Didn't we switch to primary?

Reply:
syntax/MultiApps/index.md still uses main and sub.

Reply:
I thought we want to use main-application and sub-application?

Reply:
I think so too.

Reply:
We're switching away from the sub description as the master description goes out, but obviously that's not everywhere yet

Reply:
I thought we were (and pretty much are) getting rid of master-slave, not main-sub.

Reply:
🤷🏼 that's why I asked